data_steward/cdr_cleaner/cleaning_rules/replace_standard_id_in_domain_tables.py

```diff
@@ -38,373 +38,401 @@
-    :param sandbox_dataset_id: Identifies the sandbox dataset to store rows 
-    #TODO use sandbox_dataset_id for CR
-    :return: a list of query dicts for replacing standard_concept_ids in domain_tables
-    """
-    queries_list = []
-    queries_list.extend(
-        get_src_concept_id_logging_queries(project_id, dataset_id))
-    queries_list.extend(
-        get_src_concept_id_update_queries(project_id, dataset_id))
-    queries_list.extend(get_mapping_table_update_queries(
-        project_id, dataset_id))
-
-    return queries_list
-
-
-def parse_args():
-    """
-    This function expands the default argument list defined in cdr_cleaner.args_parser
-    :return: an expanded argument list object
-    """
-
-    import cdr_cleaner.args_parser as parser
-
-    additional_argument = {
-        parser.SHORT_ARGUMENT: '-n',
-        parser.LONG_ARGUMENT: '--snapshot_dataset_id',
-        parser.ACTION: 'store',
-        parser.DEST: 'snapshot_dataset_id',
-        parser.HELP: 'Create a snapshot of the dataset',
-        parser.REQUIRED: True
-    }
-    args = parser.default_parse_args([additional_argument])
-    return args
+SRC_CONCEPT_ID_MAPPING_QUERY = JINJA_ENV.from_string("""
+SELECT DISTINCT 
+    '{{table_name}}' AS domain_table,
+    domain.{{table_name}}_id AS src_id,
+    domain.{{table_name}}_id AS dest_id,
+    domain.{{domain_concept_id}} AS concept_id,
+    domain.{{domain_source}} AS src_concept_id,
+    coalesce(dcr.concept_id_2, scr.concept_id_2, domain.{{domain_concept_id}}) AS new_concept_id,
+    CASE
+        WHEN domain.{{domain_source}} = 0 THEN domain.{{domain_concept_id}}
+        ELSE domain.{{domain_source}}
+    END AS new_src_concept_id,
+    dcr.concept_id_2 IS NOT NULL AS lookup_concept_id, 
+    dcr.concept_id_2 IS NULL AND scr.concept_id_2 IS NOT NULL AS lookup_src_concept_id,
+    domain.{{domain_source}} = 0 AND dcr.concept_id_2 IS NOT NULL AS is_src_concept_id_replaced,
+    CASE
+        WHEN dcr.concept_id_2 IS NOT NULL THEN 'replaced using concept_id'
+        WHEN scr.concept_id_2 IS NOT NULL THEN 'replaced using source_concept_id'
+        ELSE 'kept the original concept_id'
+    END AS action
+FROM `{{project}}.{{dataset}}.{{table_name}}` AS domain 
+LEFT JOIN 
+    `{{project}}.{{dataset}}.concept` AS dc 
+ON 
+    domain.{{domain_concept_id}} = dc.concept_id
+LEFT JOIN 
+    `{{project}}.{{dataset}}.concept_relationship` AS dcr 
+ON 
+    dcr.concept_id_1 = dc.concept_id AND dcr.relationship_id = 'Maps to' 
+LEFT JOIN 
+    `{{project}}.{{dataset}}.concept` AS sc 
+ON 
+    domain.{{domain_source}} = sc.concept_id 
+LEFT JOIN 
+    `{{project}}.{{dataset}}.concept_relationship` AS scr 
+ON 
+    scr.concept_id_1 = sc.concept_id AND scr.relationship_id = 'Maps to' 
+WHERE 
+    dc.standard_concept IS NULL or dc.standard_concept = 'C'
+""")
+
+DUPLICATE_ID_UPDATE_QUERY = JINJA_ENV.from_string("""
+UPDATE 
+    `{{project}}.{{dataset}}.{{logging_table}}` AS to_update 
```

Comment:
@nishanthpp93 this is a good question. I pondered a little bit about this and I think it may not belong to `sandbox` because the purpose of the `sandbox` dataset is to store the original records that are going to be modified/deleted. This `logging_table` is different from the `sandbox` tables in that it is actually a mapping table, it guides the mapping process for the `domain_concept_id` fields so I think it should stay in the `dataset`.

Reply:
Sandbox is also used to store mapping/lookup tables in addition to original records and I think this was already discussed earlier @ChaoPang. The main reasoning for this is to allow copying the dataset and sending to downstream teams without having to remove internal tables since they don't need it.

Reply:
Okay sounds good.

Reply:
This is done